### ISSUE TITLE
Suggestion: facilitate recursivity

### ISSUE THREAD

**Issue by ngallet** [chuyskywalker/rolling-curl#32]
Hello,

I use this script since the 1.0 version for recursivity (especially with the "callback" function in adding new URL on the fly) while this version seems not to do it (I'm probably wrong and don't know how to use it as it is), so here my suggestion (what I have done).

In "rolling-curl/src/RollingCurl/RollingCurl.php", just move the "callback" (lines 275 to 279, "_remove the curl handle that just completed_") before line 262 ("_// start a new request (it's important to do this before removing the old one_").

With this, you can add new URL on the fly in the "callback" function (you have to pass "RollingCurl" object in it).

Maybe it can be changed on these source?

Ty for this very useful script.

**Comment by NickStallman** [comment]
Agreed, being able to feed additional urls in to it from the callback is pretty important.

A basic use case is a web crawler where it fetches a page, extracts the links from it and adds those links back in to the queue.
It's very useful for a lot more than just crawlers however.
